src/main/scala/com/github.sstone/amqp/ChannelOwner.scala

```diff
@@ -51,10 +60,9 @@ object ChannelOwner {
           def handleFlow(active: Boolean): Unit = listener ! HandleFlow(active)
         }))
       }
-      case request@Publish(exchange, routingKey, body, properties, mandatory, immediate) => {
+      case request@Publish(_, _, _, _, _, _) => {
```

Comment:
It's just a theoretical thing, that has no practical implications in this particular case. I would just prefer to have it as a reflex that we don't match on types, because matching on a type is like calling .isInstanceOf[] on something. If it becomes a reflex, then it won't happen in cases where it actually matters. 

(These are cases for example, when the function is parameterized, because in such cases isInstanceOf[] and typecasts break parametricity, which is a really useful property of such functions, that provides you with guarantees based on the type of the function. For example if you have an ```A => Boolan``` function, you can be sure that either it will always and only return true, or it will always and only return false, because if the only value that goes in is of type A, then you can only "create" a value of type Boolean, if you hardcode it. But if you return param.isInstanceOf[Int], then you lose this extra knowledge, that is encoded in the type of the function. This particular function is not parameterized, so parametricity is irrelevant here, and I might be wrong that getting into the habit of generally not matching on types is a good thing, because we are clever enough to distinguish, but to me it seems simpler to just avoid it. Anyway)